src/System.Management.Automation/engine/NativeCommandProcessor.cs

```diff
@@ -1146,20 +1163,59 @@ private ProcessStartInfo GetProcessStartInfo(bool redirectOutput, bool redirectE
                 {
                     // Shell doesn't exist on headless SKUs, so documents cannot be associated with an application.
                     // Therefore, we cannot run document in this case.
-                    throw InterpreterError.NewInterpreterException(this.Path, typeof(RuntimeException),
-                        this.Command.InvocationExtent, "CantActivateDocumentInPowerShellCore", ParserStrings.CantActivateDocumentInPowerShellCore, this.Path);
+                    throw InterpreterError.NewInterpreterException(
+                        this.Path,
+                        typeof(RuntimeException),
+                        this.Command.InvocationExtent,
+                        "CantActivateDocumentInPowerShellCore",
+                        ParserStrings.CantActivateDocumentInPowerShellCore,
+                        this.Path);
                 }
 
                 // We only want to ShellExecute something that is standalone...
                 if (!soloCommand)
                 {
-                    throw InterpreterError.NewInterpreterException(this.Path, typeof(RuntimeException),
-                        this.Command.InvocationExtent, "CantActivateDocumentInPipeline", ParserStrings.CantActivateDocumentInPipeline, this.Path);
+                    throw InterpreterError.NewInterpreterException(
+                        this.Path,
+                        typeof(RuntimeException),
+                        this.Command.InvocationExtent,
+                        "CantActivateDocumentInPipeline",
+                        ParserStrings.CantActivateDocumentInPipeline,
+                        this.Path);
                 }
 
                 startInfo.UseShellExecute = true;
             }
 
+            return startInfo;
+        }
+
+        /// <summary>
+        /// Get whether we should treat this executable with special handling and use the legacy passing style.
+        /// </summary>
+        /// <param name="filePath"></param>
+        private bool UseSpecialArgumentPassing(string filePath)
+        {
+            // We need to check if we're using legacy argument passing or it's a special case.
+            bool useLegacy = NativeParameterBinderController.ArgumentPassingStyle == NativeArgumentPassingStyle.Legacy;
+            bool windowsSpecialCase =
+                NativeParameterBinderController.ArgumentPassingStyle == NativeArgumentPassingStyle.Windows &&
+                UseLegacyPassingStyle(filePath);
+            return (useLegacy || windowsSpecialCase);
+        }
```

Comment:
No need evaluate UseLegacyPassingStyle() if Legacy or not Windows:
```suggestion
        private bool UseSpecialArgumentPassing(string filePath) =>
            NativeParameterBinderController.ArgumentPassingStyle == NativeArgumentPassingStyle.Legacy
            || (NativeParameterBinderController.ArgumentPassingStyle == NativeArgumentPassingStyle.Windows && UseLegacyPassingStyle(filePath));
```